subhub/api/webhooks/stripe/customer.py

```diff
@@ -59,53 +62,54 @@ def run(self):
-            "trial_start": self.payload.data.object.trial_start,
-            "trial_end": self.payload.data.object.trial_end,
-            "tax_percent": self.payload.data.object.tax_percent,
-            "application_fee_percent": self.payload.data.object.application_fee_percent,
-            "user_id": self.payload.data.object.metadata.get("userid", None),  # why?
-        }
-        routes = [StaticRoutes.SALESFORCE_ROUTE, StaticRoutes.FIREFOX_ROUTE]
-        self.send_to_routes(routes, json.dumps(data))
+        user_id = self.payload.data.object.metadata.get("userid", None)
+        updated_customer = ""
+        if not user_id:
+            customer_id = self.payload.object.customer
+            updated_customer = stripe.Customer.retrieve(customer_id)
+            user_id = updated_customer.metadata.get("userid")
+        sub_status = self.payload.data.object.status in ["active", "trialing"]
+        if user_id:
+            sfd = dict(
+                uid=user_id,
+                active=sub_status,
+                subscriptionId=self.payload.data.object.id,
+                productName=self.payload.data.object.plan.nickname,
+                eventId=self.payload.id,
+                event_id=self.payload.id,
+                eventCreatedAt=self.payload.created,
+                messageCreatedAt=int(time.time()),
+            )
+            routes = [StaticRoutes.FIREFOX_ROUTE]
+            self.send_to_routes(routes, json.dumps(sfd))
+        else:
+            raise ClientError(f"userid is None for customer {updated_customer}")
 
 
 class StripeCustomerSubscriptionDeleted(AbstractStripeWebhookEvent):
     def run(self):
-        # items is a keyword in the attrdict; this steps around that
-        items = AttrDict(self.payload.data.object["items"])
-        data = {
-            "event_id": self.payload.id,
-            "event_type": self.payload.type,
-            "customer_id": self.payload.data.object.customer,
-            "subscription_id": self.payload.data.object.id,
-            "current_period_start": self.payload.data.object.current_period_start,
-            "current_period_end": self.payload.data.object.current_period_end,
-            "subscription_created": items.data[0].created,
-            "plan_amount": self.payload.data.object.plan.amount,
-            "plan_currency": self.payload.data.object.plan.currency,
-            "plan_name": self.payload.data.object.plan.nickname,
-            "trial_period_days": self.payload.data.object.plan.trial_period_days,
-            "status": self.payload.data.object.status,
-            "canceled_at": self.payload.data.object.canceled_at,
-            "created": self.payload.data.object.created,
-        }
-        routes = [StaticRoutes.SALESFORCE_ROUTE, StaticRoutes.FIREFOX_ROUTE]
-        self.send_to_routes(routes, json.dumps(data))
+        user_id = self.payload.data.object.metadata.get("userid", None)
+        updated_customer = ""
```

Comment:
why are we setting this to the empty string?

Reply:
Good catch, changed to self.payload.object.customer in the error log below.